ProcessList.c

```diff
@@ -354,6 +354,8 @@ void ProcessList_expandTree(ProcessList* this) {
 }
 
 void ProcessList_collapseAllBranches(ProcessList* this) {
+   ProcessList_buildTree(this); // Update `tree_depth` fields of the processes
+   this->needsSort = true; // Avoid toposort on startup with collapseAllBranches + !treeView
```

Comment:
What is "toposort"? Can you clarify the comment / the rationale?

Reply:
Elaborated the comment a bit. This is a failsafe for the case if, after some changes, `ProcessList_collapseAllBranches` starts being called in non-tree mode while the process list is already sorted. In that case `ProcessList_buildTree` would change sort order to "by effective parent PID", and the code that preserves the sort order to avoid processes jumping around would preserve this incorrect sort order. Right now this does not happen anyway because the shortcut is inactive in flat mode (2nd patch) and the initial call on startup happens before the list is sorted for the first time.